### PR TITLE
Adding the default role for IM plugin

### PR DESCRIPTION
**Pull request by thalurur**
Signed-off-by: Ravi Thaluru <ravi1092@gmail.com>

##  opensearch-security pull request intake form
_Please provide as much details as possible to get feedback/acceptance on your PR quickly_
Adding a new full access role for IndexManagement plugin so users who want to manage all indices, create policies, create rollups, create transforms can use the deafult role. 

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by hardik-k-shah — DISMISSED** [review]

**Review by hardik-k-shah — DISMISSED** [review]

File: securityconfig/roles.yml [hunk 1]
```diff
@@ -129,6 +129,38 @@ asynchronous_search_read_access:
...
+    - "cluster:admin/opendistro/ism/policy/write"
+    - "cluster:admin/opendistro/ism/policy/search"
+    - "cluster:admin/opendistro/ism/policy/get"
+    - "cluster:admin/opendistro/ism/policy/delete"
+    - "cluster:admin/opendistro/ism/managedindex/add"
+    - "cluster:admin/opendistro/ism/managedindex/remove"
+    - "cluster:admin/opendistro/ism/managedindex/change"
+    - "cluster:admin/opendistro/ism/managedindex/retry"
+    - "cluster:admin/opendistro/ism/managedindex/explain"
+    - "cluster:admin/opendistro/rollup/index"
```
Comment (hardik-k-shah) [inline securityconfig/roles.yml:145]: shouldn't these rollup and transform be a part of ISM plugin and should have same path as of policy and mangedindex ?

ideally it should be like .. ```ism/rollup``` and ```ism/transform``` same as ```ism/policy``` and ```ism/managedindex```

Comment (thalurur) [reply securityconfig/roles.yml:145]: Created an issue in IndexManagement https://github.com/opensearch-project/index-management/issues/138

File: securityconfig/roles.yml [hunk 2]
```diff
@@ -129,6 +129,38 @@ asynchronous_search_read_access:
   cluster_permissions:
     - 'cluster:admin/opendistro/asynchronous_search/get'
 
+# Allows user to use all index_management actions - ism policies, rollups, transforms
+index_management_full_access:
+  reserved: true
+  cluster_permissions:
+    - "cluster:admin/opendistro/ism/policy/write"
```
Comment (vrozov) [inline securityconfig/roles.yml:136]: Use wildcard: `cluster:admin/opendistro/ism/*`

**Comment by vrozov** [comment]
rebase against `main` and sign commit.

File: securityconfig/roles.yml (hunk 2 above)
Comment (thalurur) [reply securityconfig/roles.yml]: updated, did the same for rollup and transform

File: securityconfig/roles.yml [hunk 3]
```diff
@@ -129,6 +129,19 @@ asynchronous_search_read_access:
...
+  reserved: true
+  cluster_permissions:
+    - "cluster:admin/opendistro/ism/*"
+    - "cluster:admin/opendistro/rollup/*"
+    - "cluster:admin/opendistro/transform/*"
+  index_permissions:
+    - index_patterns:
+        - '*'
+      allowed_actions:
+        - 'indices:admin/opensearch/ism/managedindex'
```
Comment (hardik-k-shah) [inline securityconfig/roles.yml:143]: here also you can use wildcard .. so in future if ism introduce more action for indices, we won't need to update this full access role.

Comment (thalurur) [reply securityconfig/roles.yml]: done

**Review by vrozov — APPROVED** [review]

**Review by hardik-k-shah — APPROVED** [review]

**Pull request merged**